src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -394,12 +395,13 @@ module.exports = function(
     if (instance.state !== oldState) {
       if (__DEV__) {
         warning(
-          false,
+          didWarnStateDeprecation,
```

Comment:
I'd like to deduplicate this by component name. (We have a `getComponentName` call below that can be extracted into a variable and then used as key.)

Reply:
I have changed this to getComponentName(..) || 'Component'. This is because, I was getting a flow error in this line::: !!stateDeprecationWarning[currentComponent], which said
>>> access of computed property/element. Computed property cannot be accessed with null
Not sure if giving default value is the right fix for this.